confgenerator/feature.go

```diff
@@ -0,0 +1,176 @@
+			features, err := trackingFeatures(p, MetricsProcessorTypes.Subagent, MetricsProcessorTypes.Kind, p.Type())
+			if err != nil {
+				return nil, err
+			}
+			allFeatures = append(allFeatures, features...)
+		}
+	}
+	if uc.Logging != nil {
+		for _, r := range uc.Logging.Receivers {
+			features, err := trackingFeatures(r, LoggingReceiverTypes.Subagent, LoggingReceiverTypes.Kind, r.Type())
+			if err != nil {
+				return nil, err
+			}
+			allFeatures = append(allFeatures, features...)
+		}
+		for _, p := range uc.Logging.Processors {
+			features, err := trackingFeatures(p, LoggingProcessorTypes.Subagent, LoggingProcessorTypes.Kind, p.Type())
+			if err != nil {
+				return nil, err
+			}
+			allFeatures = append(allFeatures, features...)
+		}
+	}
+
+	return allFeatures, nil
+}
+
+func trackingFeatures(component interface{}, module, kind, typ string) ([]Feature, error) {
+	return trackingFeaturesHelper(component, module, kind, typ, "")
+}
+
+func trackingFeaturesHelper(component any, module, kind, typ, prefix string) ([]Feature, error) {
+	if component == nil {
+		return []Feature{}, nil
+	}
+	c := reflect.ValueOf(component)
+	t := c.Type()
+
+	if t.Kind() == reflect.Ptr {
+		t = t.Elem()
+	}
+
+	var features []Feature
+	// Iterate over all available fields and read the tag value
+	for i := 0; i < t.NumField(); i++ {
+		// Get the field, returns https://golang.org/pkg/reflect/#StructField
+		field := t.Field(i)
+
+		// Grab actual value instead of pointer
+		v := reflect.Indirect(reflect.Indirect(c).FieldByName(field.Name))
+		if v.IsZero() {
+			continue
+		}
+
+		// Get the field tag value
+		trackingTag := field.Tag.Get("tracking")
+		tags := strings.Split(trackingTag, ",")
+		trackingName := tags[0]
+
+		// Check to see if value needs to be overridden
+		var overridden string
+		if len(tags) > 1 {
+			overridden = tags[1]
+		}
+		yamlTag := field.Tag.Get("yaml")
+		isInline := strings.Contains(yamlTag, ",inline")
+
+		// Append prefix with dot notation for child features key
+		key := fmt.Sprintf("%s.%s", prefix, trackingName)
+		if prefix == "" || isInline {
+			key = trackingName
+		}
+
+		if v.Type().Kind() == reflect.Struct {
+			if isInline && trackingTag != "" {
+				return nil, ErrTrackingInlineStruct
+			}
+			tf, err := trackingFeaturesHelper(v.Interface(), module, kind, typ, key)
+			if err != nil {
+				return nil, err
+			}
+			features = append(features, tf...)
+		} else {
+			if trackingTag == "" {
+				continue
+			}
+			featureTrackingString, err := asString(v.Interface())
+			if err != nil {
+				return nil, err
+			}
+			if overridden != "" {
+				featureTrackingString = overridden
+			}
+			features = append(features, Feature{
+				Module: module,
+				Kind:   kind,
+				Type:   typ,
+				Key:    key,
+				Value:  featureTrackingString,
+			})
+		}
+	}
+
+	return features, nil
+}
+
+func asString(v interface{}) (string, error) {
+	var s string
+	switch x := v.(type) {
+	case bool:
+		s = fmt.Sprintf("%t", x)
+	case int:
```

Comment:
I think we have a couple config taking int64 - would it make sense to have this one here as `case Int, Int8, Int16, Int32, Int64:`?

Reply:
Changed in: [0f93735](https://github.com/GoogleCloudPlatform/ops-agent/pull/853/commits/0f937356b0edf460a8aa421e4e0398dfc4c018f8)